upgrade.ts

```diff
@@ -0,0 +1,127 @@
+import { resolveDirectiveNameFromSelector } from './src/facade/lang';
+import { PrecompiledDirectiveFactory } from './src/core/util/bundler';
+import { createBootstrapFn } from './src/platform/browser_utils';
+import { ProviderLiteral } from './src/core/di/provider_util';
+import { reflector } from './src/core/reflection/reflection';
+
+/**
+ * `UgradeAdapterRef` controls a hybrid AngularJS v1 / Angular v2 application,
+ * but we don't have a use for it right now so no point in creating an interface for it...
+ */
+export type UgradeAdapterRef = void;
+
+export interface UpgradeAdapter {
+  new (): upgradeAdapterInstance;
+}
+
+export interface upgradeAdapterInstance {
+  /**
+   * Allows Angular v2 Component to be used from AngularJS v1.
+   */
+  downgradeNg2Component(type: Type): Function;
+  /**
+   * Bootstrap a hybrid AngularJS v1 / Angular v2 application.
+   */
+  bootstrap(element: Element, modules?: any[], config?: angular.IAngularBootstrapConfig): UgradeAdapterRef;
+  /**
+   * Adds a provider to the top level environment of a hybrid AngularJS v1 / Angular v2 application.
+   */
+  addProvider(provider: Type | any[] | any): void;
+  /**
+   * Allows Angular v2 service to be accessible from AngularJS v1.
+   */
+  downgradeNg2Provider(token: any): Function;
+}
+
+export class NgMetadataUpgradeAdapter {
```

Comment:
I would keep the naming like ng2/upgrade has.

so just use `export class UpgradeAdapter {}`

and then within main it will be clearly separated by import path:

```typescript
import {UpgradeAdapter as Ng2UpgradeAdapter} from '@angular/upgrade';
import {UpgradeAdapter} from 'ng-metadata/upgrade';

import {AppComponent} from './index';

const upgradeAdapter = new UpgradeAdapter(Ng2UpgradeAdapter);

upgradeAdapter.bootstrap(AppComponent,[]);
```